### DISCUSSION TITLE
Seeking Assistance with Particle-Wall Collision Simulation

### DISCUSSION THREAD

**Discussion by idop11** [glotzerlab/hoomd-blue#1752] [Support] [answered]
Hello,

I am working with a system of active particles undergoing rotational diffusion in a 2-D box. I aim to introduce four walls, one for each edge of the square box, to induce elastic collisions (or any suitable repulsive force, if elastic collisions aren't feasible) upon particle-wall interaction. My approach involved creating a plane for each of the four sides of the box using hoomd.wall.Plane, followed by applying a wall force (details are in the attached code).

However, I've encountered two primary issues in my attempts to simulate a repulsive force upon particle-wall collision:

1. In all cases, the particles reached one of the four walls and adhered to it, subsequently moving parallel to the wall. Despite experimenting with various forces (e.g., ForceShiftedLJ, LJ) and parameters (varying values of $\epsilon, \sigma, r_{cut}$), the outcome invariably resulted in the particles sticking to the walls (illustrated in the attached pictures).
2. An error likely related to a particle being initialized too close to an edge, experiencing an excessively high potential, as shown below:
```
Traceback (most recent call last):
  File "/mnt/c/Users/User/PycharmProjects/graphicsExperiments/try1.py", line 92, in <module>
    simulation.run(N_steps)
  File "/home/idop/miniconda3/envs/hoomd_newest/lib/python3.11/site-packages/hoomd/simulation.py", line 562, in run
    self._cpp_sys.run(steps_int, write_at_start)
RuntimeError: Particle with unique tag 8 is no longer in the simulation box.

Cartesian coordinates:
x: 192.991 y: 10.5626 z: 0
Fractional coordinates:
f.x: 3.9432 f.y: 0.68845 f.z: 0
Local box lo: (-28.025, -28.025, -0)
          hi: (28.025, 28.025, 0)
```
I have included a picture of the simulation at the moment this error occurred, with particle 8 highlighted.

Could anyone offer guidance on how to address these two issues? Specifically, how can the wall exert a repulsive force, and how can we avoid initialization issues with particles too close to the walls?

My script: 
[sample_code.txt](https://github.com/glotzerlab/hoomd-blue/files/14736057/sample_code.txt)

For the first situation, here are two pictures showing particles stuck to and moving along the walls (as indicated by the arrows):
<img width="320" alt="simulation1" src="https://github.com/glotzerlab/hoomd-blue/assets/127539588/dd40f3c7-36be-4890-a67a-6e0817372010">

<img width="320" alt="simulation2" src="https://github.com/glotzerlab/hoomd-blue/assets/127539588/86221c08-5db0-4e33-ba04-acf7c7324b27">

And for the second situation (initialization error), here is the picture:
<img width="320" alt="simulation3" src="https://github.com/glotzerlab/hoomd-blue/assets/127539588/4252b5b8-e836-4082-8958-8beb440a7379">

Thank you for any help or insights you can provide!
Ido

**Comment by joaander** [comment, marked as answer]
1. Your LJ wall `lj.params['A'] = {"sigma": 1.0, "epsilon": 1.0, "r_cut": 2.5}` has an attractive well. Use `r_cut = 2**(1/6)` to include only the repulsive part of the potential. To achieve elastic collisions, you must use a NVE integrator an no active force. Your active Brownian particle will continue pushing into the wall until the rotational diffusion rotates the active force away from the wall.

4. You must place particles such that the initial force on them is reasonable.

**Reply by idop11** [reply]
Thanks!
